src/SettingsMigrator/Migrator.cs

```diff
@@ -239,10 +238,10 @@ private static TestSettingsNodes ReadTestSettingsNodes(string testSettingsPath)
     /// <param name="newXmlDoc">Xml doc to process</param>
     private static void RemoveEmbeddedTestSettings(XmlDocument newXmlDoc)
     {
-        var testSettingsNode = newXmlDoc.DocumentElement.SelectSingleNode(@"/RunSettings/MSTest/SettingsFile");
+        var testSettingsNode = newXmlDoc?.DocumentElement?.SelectSingleNode(@"/RunSettings/MSTest/SettingsFile");
```

Comment:
`newXmlDoc` cannot be null.

Reply:
Can you please review all nullability in this class.

Reply:
I did. If I remove marks causes might be null warnings.